src/Components/v2/CarouselV2.tsx

```diff
@@ -0,0 +1,265 @@
+  pageDots?: boolean
+}
+
+type Arrow = (
+  props: {
+    Arrow: React.ReactType
+    getFlickity: any
+  } & Props
+) => React.ReactNode
+
+interface Props {
+  height?: number
+  data: object[] // This is designed to handle any shape of data passed, as long as its an array
+  render: (slide) => ReactNode
+  renderLeftArrow?: Arrow
+  renderRightArrow?: Arrow
+  onArrowClick?: () => void
+  settings?: FlickitySettings
+}
+
+export class Carousel extends React.Component<Props> {
+  static defaultProps = {
+    height: 300,
+  }
+
+  render() {
+    return (
+      <>
+        <Media at="xs">
+          <SmallCarousel {...this.props} />
+        </Media>
+        <Media greaterThan="xs">
+          <LargeCarousel {...this.props} />
+        </Media>
+      </>
+    )
+  }
+}
+
+/*
+ * Returns null if no window for SSR
+ */
+const Flickity =
+  typeof window !== "undefined"
+    ? require("react-flickity-component")
+    : () => null
+
+export const LargeCarousel = (props: Props) => {
+  const [isMounted, toggleMounted] = useState(false)
+  const [currentSlideIndex, setSlideIndex] = useState(0)
+  const [lastItemVisible, setLastItemVisible] = useState(false)
+
+  useEffect(() => {
+    toggleMounted(true)
+  }, [])
+  const flicktyRef = useRef(null)
+
+  let flickity = null
+
+  const checkLastItemVisible = () => {
+    if (!flickity.slides.length) {
+      return
+    }
+    const isLastItemVisible = flickity.selectedElements.includes(
+      flickity.getLastCell().element
+    )
+    setLastItemVisible(isLastItemVisible)
+  }
+
+  if (isMounted) {
+    if (flicktyRef.current !== null) {
+      flickity = flicktyRef.current.flkty
+    }
+
+    flickity.on("select", index => {
+      setSlideIndex(index)
+      checkLastItemVisible()
+    })
+  }
+
+  const settings = {
+    draggable: false,
+    freeScroll: false,
+    wrapAround: false,
+    cellAlign: "left",
+    pageDots: false,
+    prevNextButtons: false,
+    groupCells: true,
+    contain: true,
+    ...props.settings,
+  }
+
+  const LeftArrow = () => {
+    const getFlickity = flickity
+    return (
+      <ArrowButton
+        onClick={() => {
+          getFlickity && getFlickity.previous ? getFlickity.previous() : null // check existence for tests
+        }}
+      >
+        <ChevronIcon direction="left" fill="black100" width={30} height={30} />
+      </ArrowButton>
+    )
+  }
+
+  const RightArrow = () => {
+    const getFlickity = flickity
+    return (
+      <ArrowButton
+        onClick={() => {
+          getFlickity && getFlickity.next ? getFlickity.next() : null // check existence for tests
+        }}
+      >
+        <ChevronIcon direction="right" fill="black100" width={30} height={30} />
+      </ArrowButton>
+    )
+  }
+  return (
+    <Flex
+      flexDirection="row"
+      position="relative"
+      justifyContent="space-around"
+      alignItems="center"
+      height={props.height}
+    >
+      {(currentSlideIndex !== 0 || settings.wrapAround === true) && (
```

Comment:
if `currentSlideIndex !== 0 || settings.wrapAround === true` means "showArrow", then best to assign this to a variable just above the `return` so that future devs don't need to parse logic to understand behavior at a glance.